### PR TITLE
Login branding

### PR DESCRIPTION
**Pull request by mario**
Adds support for the following server url structures:

*.whatever.com
whatever.com/*

### CODE REVIEW

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 1]
```diff
@@ -436,20 +441,30 @@ private void initServerPreFragment(Bundle savedInstanceState) {
...
             mHostUrlInput.setEnabled(false);
             mHostUrlInput.setFocusable(false);
         }
+
+        String serverInputType = getResources().getString(R.string.server_input_type);
+
         if (isUrlInputAllowed) {
             mRefreshButton = findViewById(R.id.embeddedRefreshButton);
+            if (mAction == ACTION_CREATE) {
+                if (serverInputType.equals(DIRECTORY_SERVER_INPUT_TYPE) ||
```
Comment (mario) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Looks significantly nicer this way, so please let me have it - thank you! :)

File: src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 2]
```diff
@@ -438,20 +443,30 @@ private void initServerPreFragment(Bundle savedInstanceState) {
...
             mHostUrlInput.setEnabled(false);
             mHostUrlInput.setFocusable(false);
         }
+
+        String serverInputType = getResources().getString(R.string.server_input_type);
+
         if (isUrlInputAllowed) {
             mRefreshButton = findViewById(R.id.embeddedRefreshButton);
+            if (mAction == ACTION_CREATE) {
+                if (serverInputType.equals(DIRECTORY_SERVER_INPUT_TYPE) ||
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java]: @mario regarding your comment that this looks nicer, I think the fix looks basically the same ;)

actual:
```
if (mAction == ACTION_CREATE) {
                if (serverInputType.equals(DIRECTORY_SERVER_INPUT_TYPE) ||
                serverInputType.equals(SUBDOMAIN_SERVER_INPUT_TYPE)) {
```

Fixed:
```
if (mAction == ACTION_CREATE) &&
                (serverInputType.equals(DIRECTORY_SERVER_INPUT_TYPE) ||
                serverInputType.equals(SUBDOMAIN_SERVER_INPUT_TYPE)) {
```

So I'd say we fix it :D What do you think?

**Comment by mario** [comment]
@AndyScherzinger think you can review this today (Monday?) code-wise so we can merge it by tomorrow? Thanks!

**Comment by AndyScherzinger** [comment]
@mario could wrap it up yesterday, and yes that's the plan. So I'll see that I get this bagged and tagged today :)

**Comment by AndyScherzinger** [comment]
👍 code wise and also tested the config values, waiting for drone and codacy to finish. Merging afterwards with single approval due to vacation situation

**Pull request merged**